kubetest/aks.go

```diff
@@ -0,0 +1,297 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"os/exec"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-10-01/containerservice"
+	"github.com/Azure/go-autorest/autorest/azure"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/crypto/ssh"
+)
+
+type aksDeployer struct {
+	azureCreds    *Creds
+	azureClient   *AzureClient
+	templateUrl   string
+	outputDir     string
+	resourceGroup string
+	resourceName  string
+}
+
+func newAksDeployer() (*aksDeployer, error) {
+	if err := validateAksFlags(); err != nil {
+		return nil, err
+	}
+
+	creds, err := getAzCredentials()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get azure credentials: %v", err)
+	}
+	env, err := azure.EnvironmentFromName(*aksAzureEnv)
```

Comment:
just double-checking, this is still in the aks-engine deployer and i guess this is fine for the PR purposes and the goal is to split into a common Azure or even just define them locally in `aks.go` based on what we need later?

Reply:
I think i'd like to split the flags and share a common azure set. Since you'll never invoke with two deployers, I didn't think it was a high priority fix. Something frustrating is that the flags are mostly used directly from different parts of the code, not passed down, which makes it a slightly more involved change, but nothing crazy.